dipy/reconst/dti.py

```diff
@@ -740,11 +741,12 @@ def __init__(self, gtab, fit_method="WLS", *args, **kwargs):
 
         if not callable(fit_method):
             try:
-                self.fit_method = common_fit_methods[fit_method]
+                fit_method = common_fit_methods[fit_method]
             except KeyError:
                 raise ValueError('"' + str(fit_method) + '" is not a known fit '
                                  'method, the fit method should either be a '
                                  'function or one of the common fit methods')
+        self.fit_method = fit_method
```

Comment:
Nice catch! Would you mind adding a test that fails without this change, so that we don't end up reverting without noticing?